hack/verify-api-resource-list.sh

```diff
@@ -0,0 +1,68 @@
+#!/bin/bash
+
+# Copyright 2018 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Script to fetch latest api resources served by kube apiserver
+# Puts the updated list at test/e2e/testing-manifests/apiresource/resources_all.csv
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+KUBE_ROOT=$(dirname "${BASH_SOURCE}")/..
+source "${KUBE_ROOT}/hack/lib/init.sh"
+
+kube::golang::setup_env
+kube::etcd::install
+export PATH="${KUBE_ROOT}/third_party/etcd:${PATH}"
+
+make -C "${KUBE_ROOT}" WHAT=cmd/kube-apiserver
+
+apiserver=$(kube::util::find-binary "kube-apiserver")
+
+LISTROOT="${KUBE_ROOT}/test/e2e/testing-manifests/apiresource"
+TMP_LISTROOT="${KUBE_ROOT}/_tmp/apiresource"
+_tmp="${KUBE_ROOT}/_tmp"
+
+mkdir -p "${_tmp}"
+cp -a "${LISTROOT}" "${TMP_LISTROOT}"
+trap "cp -a ${TMP_LISTROOT} ${LISTROOT}/..; rm -rf ${_tmp}" EXIT SIGINT
```

Comment:
don't we have a general trap mechanism? This command overrides any trap installed before.

Reply:
Updated to use util function to add trap